convert/github/convert.go

```diff
@@ -107,23 +117,57 @@ func (d *Converter) ConvertFile(p string) ([]byte, error) {
 	return d.Convert(f)
 }
 
-// converts converts a github pipeline pipeline.
-func (d *Converter) convert() ([]byte, error) {
+// converts a GitHub pipeline to Harness pipeline.
+func (d *Converter) convert(ctx *context) ([]byte, error) {
 
 	// create the harness pipeline
 	pipeline := &harness.Pipeline{
 		Version: 1,
-		// Default: convertDefault(d.config),
+		Stages:  []*harness.Stage{},
 	}
 
-	// for _, steps := range d.config.Pipelines.Default {
-	// 	if steps.Stage != nil {
-	// 		// TODO support for fast-fail
-	// 		d.stage = steps.Stage // push the stage to the state
-	// 		stage := d.convertStage()
-	// 		pipeline.Stages = append(pipeline.Stages, stage)
-	// 	}
-	// }
+	for _, from := range ctx.pipeline {
+		if from == nil {
+			continue
+		}
+		pipeline.Name = from.Name
+
+		if from.Jobs != nil {
+			for name, job := range from.Jobs {
+				actionJob := &job
+				var cloneStage *harness.CloneStage
+				if actionJob != nil {
+					for _, step := range actionJob.Steps {
+						cloneStage = convertClone(step)
+						if cloneStage != nil {
+							break
+						}
+					}
+				}
+
+				pipeline.Stages = append(pipeline.Stages, &harness.Stage{
+					Name:     name,
+					Type:     "ci",
+					When:     convertCond(&from.On),
```

Comment:
Instead of `&from.On` I would recommend making `On` a pointer in `Pipeline`

```diff
type Pipeline struct {
	Name        string            `yaml:"name,omitempty"`
-	On          WorkflowTriggers  `yaml:"on,omitempty"`
+	On          *WorkflowTriggers  `yaml:"on,omitempty"`
	Jobs        map[string]Job    `yaml:"jobs,omitempty"`
	Environment map[string]string `yaml:"env,omitempty"`
}
```